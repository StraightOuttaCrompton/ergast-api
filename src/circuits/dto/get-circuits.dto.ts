import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";

export class GetCircuitsDto {
    /**
     * TODO: extract into class which this inherits from
     */
    @IsOptional()
    @IsInt()
    @IsPositive()
    offset: number;

    @IsOptional()
    @IsInt()
    @IsPositive()
    limit: number;
    /**
     * --------------------------------------------------
     */

    @IsOptional()
    @IsNotEmpty()
    driverId: string;

    @IsOptional()
    @IsNotEmpty()
    constructorId: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status: number; // TODO: validate for actual status values
}
