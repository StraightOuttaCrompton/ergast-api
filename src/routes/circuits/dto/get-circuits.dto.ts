import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetCircuitsDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    driverId: string;

    @IsOptional()
    @IsNotEmpty()
    constructorId: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status: number; // TODO: validate for actual status values
}
