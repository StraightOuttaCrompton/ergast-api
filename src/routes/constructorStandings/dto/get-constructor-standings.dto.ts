import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetConstructorStandingsDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    constructorId?: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    position?: number;
}
