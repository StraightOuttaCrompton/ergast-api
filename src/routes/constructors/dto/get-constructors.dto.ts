import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetConstructorsDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    circuitId?: string;

    @IsOptional()
    @IsNotEmpty()
    driverId?: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status?: number; // TODO: validate for actual status values

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    constructorStandings?: number;
}
