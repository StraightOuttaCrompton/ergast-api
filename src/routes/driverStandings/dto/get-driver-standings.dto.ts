import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetDriverStandingsDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    driverId?: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    position?: number;
}
