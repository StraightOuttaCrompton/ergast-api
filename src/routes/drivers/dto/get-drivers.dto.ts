import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetDriversDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    circuitId?: string;

    @IsOptional()
    @IsNotEmpty()
    constructorId?: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    driverStandings?: number;
}
