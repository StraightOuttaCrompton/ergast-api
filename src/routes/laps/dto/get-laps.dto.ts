import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetLapsDto extends CommonDto {
    @Type(() => Number)
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsInt()
    @IsPositive()
    round: number;

    @IsOptional()
    @IsNotEmpty()
    driverId?: string;
}
