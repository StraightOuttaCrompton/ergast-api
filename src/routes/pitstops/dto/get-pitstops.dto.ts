import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetPitstopsDto extends CommonDto {
    @Type(() => Number)
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsInt()
    @IsPositive()
    round: number;

    @IsOptional()
    @IsNotEmpty()
    driverId: string;

    @IsOptional()
    @Type(() => Number)
    @IsInt()
    @IsPositive()
    lapNumber: number;
}
