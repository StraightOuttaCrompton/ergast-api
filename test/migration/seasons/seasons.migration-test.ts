import querystring from "querystring";
import request from "supertest";
import { GetSeasonsDto } from "../../../src/routes/seasons/dto/get-seasons.dto";
import { getMigrationTest } from "../migrationUtils";
import { app } from "../setup";
import path from "path";

describe("GET /seasons/${seasonId} smoke tests", () => {
    test("/seasons/1994", async () => {
        const response = await request(app.getHttpServer()).get("/seasons/1994");

        expect(response.body).toMatchSnapshot();
    });
});

describe("GET /seasons smoke tests", () => {
    const migrationTest = getMigrationTest(
        __dirname,
        path.basename(__filename),
        (response) => response.MRData.SeasonTable.Seasons
    );

    const endpoint = "/seasons";

    const queries: { queryParams: Partial<GetSeasonsDto>; legacyRoute: string }[] = [
        {
            queryParams: { driverId: "bottas" },
            legacyRoute: "drivers/bottas",
        },
        {
            queryParams: { driverId: "alonso", constructorId: "minardi" },
            legacyRoute: "drivers/alonso/constructors/minardi",
        },
        {
            queryParams: { driverId: "hamilton", constructorId: "mercedes", circuitId: "imola" },
            legacyRoute: "drivers/hamilton/constructors/mercedes/circuits/imola",
        },
        {
            queryParams: { driverId: "maldonado", status: 4 },
            legacyRoute: "drivers/maldonado/status/4",
        },
        {
            queryParams: { driverId: "vettel", constructorId: "ferrari", result: 1 },
            legacyRoute: "drivers/vettel/constructors/ferrari/results/1",
        },
        {
            queryParams: { driverId: "max_verstappen", grid: 1 },
            legacyRoute: "drivers/max_verstappen/grid/1",
        },
        {
            queryParams: { driverId: "sainz", fastest: 1 },
            legacyRoute: "drivers/sainz/fastest/1",
        },
        {
            queryParams: { driverId: "sainz", driverStandings: 5 },
            legacyRoute: "drivers/sainz/driverStandings/5",
        },
        {
            queryParams: { constructorId: "ferrari", driverStandings: 5 },
            legacyRoute: "constructors/ferrari/driverStandings/5",
        },
    ];

    migrationTest(endpoint, `https://ergast.com/api/f1/seasons.json`);

    queries.forEach(({ queryParams, legacyRoute }) => {
        migrationTest(
            `${endpoint}?${querystring.stringify(queryParams)}`,
            `https://ergast.com/api/f1/${legacyRoute}/seasons.json`
        );
    });
});
