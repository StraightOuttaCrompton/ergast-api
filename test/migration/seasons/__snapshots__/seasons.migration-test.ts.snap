// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /seasons smoke tests /seasons 1`] = `
[
  {
    "season": "1950",
    "url": "http://en.wikipedia.org/wiki/1950_Formula_One_season",
  },
  {
    "season": "1951",
    "url": "http://en.wikipedia.org/wiki/1951_Formula_One_season",
  },
  {
    "season": "1952",
    "url": "http://en.wikipedia.org/wiki/1952_Formula_One_season",
  },
  {
    "season": "1953",
    "url": "http://en.wikipedia.org/wiki/1953_Formula_One_season",
  },
  {
    "season": "1954",
    "url": "http://en.wikipedia.org/wiki/1954_Formula_One_season",
  },
  {
    "season": "1955",
    "url": "http://en.wikipedia.org/wiki/1955_Formula_One_season",
  },
  {
    "season": "1956",
    "url": "http://en.wikipedia.org/wiki/1956_Formula_One_season",
  },
  {
    "season": "1957",
    "url": "http://en.wikipedia.org/wiki/1957_Formula_One_season",
  },
  {
    "season": "1958",
    "url": "http://en.wikipedia.org/wiki/1958_Formula_One_season",
  },
  {
    "season": "1959",
    "url": "http://en.wikipedia.org/wiki/1959_Formula_One_season",
  },
  {
    "season": "1960",
    "url": "http://en.wikipedia.org/wiki/1960_Formula_One_season",
  },
  {
    "season": "1961",
    "url": "http://en.wikipedia.org/wiki/1961_Formula_One_season",
  },
  {
    "season": "1962",
    "url": "http://en.wikipedia.org/wiki/1962_Formula_One_season",
  },
  {
    "season": "1963",
    "url": "http://en.wikipedia.org/wiki/1963_Formula_One_season",
  },
  {
    "season": "1964",
    "url": "http://en.wikipedia.org/wiki/1964_Formula_One_season",
  },
  {
    "season": "1965",
    "url": "http://en.wikipedia.org/wiki/1965_Formula_One_season",
  },
  {
    "season": "1966",
    "url": "http://en.wikipedia.org/wiki/1966_Formula_One_season",
  },
  {
    "season": "1967",
    "url": "http://en.wikipedia.org/wiki/1967_Formula_One_season",
  },
  {
    "season": "1968",
    "url": "http://en.wikipedia.org/wiki/1968_Formula_One_season",
  },
  {
    "season": "1969",
    "url": "http://en.wikipedia.org/wiki/1969_Formula_One_season",
  },
  {
    "season": "1970",
    "url": "http://en.wikipedia.org/wiki/1970_Formula_One_season",
  },
  {
    "season": "1971",
    "url": "http://en.wikipedia.org/wiki/1971_Formula_One_season",
  },
  {
    "season": "1972",
    "url": "http://en.wikipedia.org/wiki/1972_Formula_One_season",
  },
  {
    "season": "1973",
    "url": "http://en.wikipedia.org/wiki/1973_Formula_One_season",
  },
  {
    "season": "1974",
    "url": "http://en.wikipedia.org/wiki/1974_Formula_One_season",
  },
  {
    "season": "1975",
    "url": "http://en.wikipedia.org/wiki/1975_Formula_One_season",
  },
  {
    "season": "1976",
    "url": "http://en.wikipedia.org/wiki/1976_Formula_One_season",
  },
  {
    "season": "1977",
    "url": "http://en.wikipedia.org/wiki/1977_Formula_One_season",
  },
  {
    "season": "1978",
    "url": "http://en.wikipedia.org/wiki/1978_Formula_One_season",
  },
  {
    "season": "1979",
    "url": "http://en.wikipedia.org/wiki/1979_Formula_One_season",
  },
]
`;

exports[`GET /seasons smoke tests /seasons?constructorId=ferrari&driverStandings=5 1`] = `
[
  {
    "season": "1958",
    "url": "http://en.wikipedia.org/wiki/1958_Formula_One_season",
  },
  {
    "season": "1959",
    "url": "http://en.wikipedia.org/wiki/1959_Formula_One_season",
  },
  {
    "season": "1960",
    "url": "http://en.wikipedia.org/wiki/1960_Formula_One_season",
  },
  {
    "season": "1961",
    "url": "http://en.wikipedia.org/wiki/1961_Formula_One_season",
  },
  {
    "season": "1962",
    "url": "http://en.wikipedia.org/wiki/1962_Formula_One_season",
  },
  {
    "season": "1963",
    "url": "http://en.wikipedia.org/wiki/1963_Formula_One_season",
  },
  {
    "season": "1964",
    "url": "http://en.wikipedia.org/wiki/1964_Formula_One_season",
  },
  {
    "season": "1965",
    "url": "http://en.wikipedia.org/wiki/1965_Formula_One_season",
  },
  {
    "season": "1966",
    "url": "http://en.wikipedia.org/wiki/1966_Formula_One_season",
  },
  {
    "season": "1967",
    "url": "http://en.wikipedia.org/wiki/1967_Formula_One_season",
  },
  {
    "season": "1968",
    "url": "http://en.wikipedia.org/wiki/1968_Formula_One_season",
  },
  {
    "season": "1969",
    "url": "http://en.wikipedia.org/wiki/1969_Formula_One_season",
  },
  {
    "season": "1970",
    "url": "http://en.wikipedia.org/wiki/1970_Formula_One_season",
  },
  {
    "season": "1971",
    "url": "http://en.wikipedia.org/wiki/1971_Formula_One_season",
  },
  {
    "season": "1972",
    "url": "http://en.wikipedia.org/wiki/1972_Formula_One_season",
  },
  {
    "season": "1973",
    "url": "http://en.wikipedia.org/wiki/1973_Formula_One_season",
  },
  {
    "season": "1974",
    "url": "http://en.wikipedia.org/wiki/1974_Formula_One_season",
  },
  {
    "season": "1975",
    "url": "http://en.wikipedia.org/wiki/1975_Formula_One_season",
  },
  {
    "season": "1976",
    "url": "http://en.wikipedia.org/wiki/1976_Formula_One_season",
  },
  {
    "season": "1977",
    "url": "http://en.wikipedia.org/wiki/1977_Formula_One_season",
  },
  {
    "season": "1978",
    "url": "http://en.wikipedia.org/wiki/1978_Formula_One_season",
  },
  {
    "season": "1979",
    "url": "http://en.wikipedia.org/wiki/1979_Formula_One_season",
  },
  {
    "season": "1980",
    "url": "http://en.wikipedia.org/wiki/1980_Formula_One_season",
  },
  {
    "season": "1981",
    "url": "http://en.wikipedia.org/wiki/1981_Formula_One_season",
  },
  {
    "season": "1982",
    "url": "http://en.wikipedia.org/wiki/1982_Formula_One_season",
  },
  {
    "season": "1983",
    "url": "http://en.wikipedia.org/wiki/1983_Formula_One_season",
  },
  {
    "season": "1984",
    "url": "http://en.wikipedia.org/wiki/1984_Formula_One_season",
  },
  {
    "season": "1985",
    "url": "http://en.wikipedia.org/wiki/1985_Formula_One_season",
  },
  {
    "season": "1986",
    "url": "http://en.wikipedia.org/wiki/1986_Formula_One_season",
  },
  {
    "season": "1987",
    "url": "http://en.wikipedia.org/wiki/1987_Formula_One_season",
  },
]
`;

exports[`GET /seasons smoke tests /seasons?driverId=alonso&constructorId=minardi 1`] = `
[
  {
    "season": "2001",
    "url": "http://en.wikipedia.org/wiki/2001_Formula_One_season",
  },
]
`;

exports[`GET /seasons smoke tests /seasons?driverId=bottas 1`] = `
[
  {
    "season": "2013",
    "url": "http://en.wikipedia.org/wiki/2013_Formula_One_season",
  },
  {
    "season": "2014",
    "url": "http://en.wikipedia.org/wiki/2014_Formula_One_season",
  },
  {
    "season": "2015",
    "url": "http://en.wikipedia.org/wiki/2015_Formula_One_season",
  },
  {
    "season": "2016",
    "url": "http://en.wikipedia.org/wiki/2016_Formula_One_season",
  },
  {
    "season": "2017",
    "url": "http://en.wikipedia.org/wiki/2017_Formula_One_season",
  },
  {
    "season": "2018",
    "url": "http://en.wikipedia.org/wiki/2018_Formula_One_World_Championship",
  },
  {
    "season": "2019",
    "url": "http://en.wikipedia.org/wiki/2019_Formula_One_World_Championship",
  },
  {
    "season": "2020",
    "url": "http://en.wikipedia.org/wiki/2020_Formula_One_World_Championship",
  },
  {
    "season": "2021",
    "url": "http://en.wikipedia.org/wiki/2021_Formula_One_World_Championship",
  },
  {
    "season": "2022",
    "url": "http://en.wikipedia.org/wiki/2022_Formula_One_World_Championship",
  },
  {
    "season": "2023",
    "url": "https://en.wikipedia.org/wiki/2023_Formula_One_World_Championship",
  },
]
`;

exports[`GET /seasons smoke tests /seasons?driverId=hamilton&constructorId=mercedes&circuitId=imola 1`] = `
[
  {
    "season": "2020",
    "url": "http://en.wikipedia.org/wiki/2020_Formula_One_World_Championship",
  },
  {
    "season": "2021",
    "url": "http://en.wikipedia.org/wiki/2021_Formula_One_World_Championship",
  },
  {
    "season": "2022",
    "url": "http://en.wikipedia.org/wiki/2022_Formula_One_World_Championship",
  },
]
`;

exports[`GET /seasons smoke tests /seasons?driverId=maldonado&status=4 1`] = `
[
  {
    "season": "2012",
    "url": "http://en.wikipedia.org/wiki/2012_Formula_One_season",
  },
  {
    "season": "2013",
    "url": "http://en.wikipedia.org/wiki/2013_Formula_One_season",
  },
  {
    "season": "2015",
    "url": "http://en.wikipedia.org/wiki/2015_Formula_One_season",
  },
]
`;

exports[`GET /seasons smoke tests /seasons?driverId=max_verstappen&grid=1 1`] = `
[
  {
    "season": "2019",
    "url": "http://en.wikipedia.org/wiki/2019_Formula_One_World_Championship",
  },
  {
    "season": "2020",
    "url": "http://en.wikipedia.org/wiki/2020_Formula_One_World_Championship",
  },
  {
    "season": "2021",
    "url": "http://en.wikipedia.org/wiki/2021_Formula_One_World_Championship",
  },
  {
    "season": "2022",
    "url": "http://en.wikipedia.org/wiki/2022_Formula_One_World_Championship",
  },
  {
    "season": "2023",
    "url": "https://en.wikipedia.org/wiki/2023_Formula_One_World_Championship",
  },
]
`;

exports[`GET /seasons smoke tests /seasons?driverId=sainz&driverStandings=5 1`] = `
[
  {
    "season": "2021",
    "url": "http://en.wikipedia.org/wiki/2021_Formula_One_World_Championship",
  },
  {
    "season": "2022",
    "url": "http://en.wikipedia.org/wiki/2022_Formula_One_World_Championship",
  },
]
`;

exports[`GET /seasons smoke tests /seasons?driverId=sainz&fastest=1 1`] = `
[
  {
    "season": "2020",
    "url": "http://en.wikipedia.org/wiki/2020_Formula_One_World_Championship",
  },
  {
    "season": "2022",
    "url": "http://en.wikipedia.org/wiki/2022_Formula_One_World_Championship",
  },
]
`;

exports[`GET /seasons smoke tests /seasons?driverId=vettel&constructorId=ferrari&result=1 1`] = `
[
  {
    "season": "2015",
    "url": "http://en.wikipedia.org/wiki/2015_Formula_One_season",
  },
  {
    "season": "2017",
    "url": "http://en.wikipedia.org/wiki/2017_Formula_One_season",
  },
  {
    "season": "2018",
    "url": "http://en.wikipedia.org/wiki/2018_Formula_One_World_Championship",
  },
  {
    "season": "2019",
    "url": "http://en.wikipedia.org/wiki/2019_Formula_One_World_Championship",
  },
]
`;

exports[`GET /seasons/\${seasonId} smoke tests /seasons/1994 1`] = `
{
  "season": "1994",
  "url": "http://en.wikipedia.org/wiki/1994_Formula_One_season",
}
`;
